contracts/SlashIndicator.sol

```diff
@@ -158,6 +188,73 @@ contract SlashIndicator is ISlashIndicator,System,IParamSubscriber, IApplication
     emit indicatorCleaned();
   }
 
+  function handleFinalityReport(FinalityEvidence calldata _evidence) external onlyInit oncePerBlock noReentrant {
+    require(
+      _evidence.numA + finalityDistance >= _evidence.numB || _evidence.numB + finalityDistance >= _evidence.numA,
+      'too long distance between blocks'
+    );
+    require(
+      _evidence.numA <= block.number &&
+      _evidence.numA + 256 >= block.number &&
+      _evidence.numB <= block.number &&
+      _evidence.numB + 256 >= block.number,
+      'block number out of range'
+    );
+
+    require(IBSCValidatorSet(VALIDATOR_CONTRACT_ADDR).isCurrentValidator(_evidence.valAddr), "not current validator");
+
+    (address[] memory vals, bytes[] memory voteAddrs) = IBSCValidatorSet(VALIDATOR_CONTRACT_ADDR).getMiningValidators();
```

Comment:
emm, we need to discuss this.  `getMiningValidators` only fetch the validator info about this epoch round, we should punish the validator even it is not in this round.

Reply:
Waiting for solution

Reply:
Add a new funciont named getLivingValidators in BSCValidatorSet.sol